DxDispatch/src/dxdispatch/Executor.cpp

```diff
@@ -633,9 +631,54 @@ void Executor::operator()(const Model::WriteFileCommand& command)
             WriteNpy(fileData, tensorType, dimensions, /*out*/ npyFileData);
             std::swap(fileDataStorage, npyFileData);
             fileData = fileDataStorage;
+
+            std::ofstream file(command.targetPath.c_str(), std::ifstream::trunc | std::ifstream::binary);
+            if (!file.is_open())
+            {
+                throw std::ios::failure("Could not open file");
+            }
+
+            file.write(reinterpret_cast<const char*>(fileData.data()), fileData.size());
+        }
+        else if (extension == ".jpg" || extension == ".png")
+        {
+            ImageTensorInfo tensorInfo = {};
+            tensorInfo.dataType = tensorType;
+            tensorInfo.channels = dimensions.size() > 1 ? dimensions[1] : 0;
+            tensorInfo.height = dimensions.size() > 2 ? dimensions[2] : 0;
+            tensorInfo.width = dimensions.size() > 3 ? dimensions[3] : 0;
```

Comment:
```suggestion
            tensorInfo.channels = dimensions.size() > 1 ? dimensions[1] : 1;
            tensorInfo.height = dimensions.size() > 2 ? dimensions[2] : 1;
            tensorInfo.width = dimensions.size() > 3 ? dimensions[3] : 1;
```
The default value for a missing dimension is 1. 0 probably won't give you the correct byte size calculation in `ParseModelBufferDesc` either.